### PR TITLE
[sensorfwd] Transfer events from worker threads via pipe

### PR DESCRIPTION
**Pull request by spiiroin**

### CODE REVIEW

File: core/hybrisadaptor.cpp [hunk 1]
```diff
@@ -1228,59 +1248,152 @@ void *HybrisManager::eventReaderThread(void *aptr)
...
+        }
 #endif
-            blockSuspend++;
+    }
+
+    /* Suspend proof sensor data processing */
+    if (wakeupEventCount)
+        ObtainTemporaryWakeLock();
+
+
```
Comment (mlehtima) [inline core/hybrisadaptor.cpp]: Nitpick, extra empty line

**Review by mlehtima — APPROVED** [review]

File: core/hybrisadaptor.cpp [hunk 2]
```diff
@@ -1228,59 +1248,152 @@ void *HybrisManager::eventReaderThread(void *aptr)
...
-        sensors_event_t *fallback = eventForHandle(data.sensor);
-        if (fallback && fallback->type == data.type && fallback->sensor == data.sensor) {
-            fallback->type = fallback->sensor = 0;
+void HybrisManager::eventPipeWakeup(int fd)
+{
+    if (m_eventPipeReadFd != fd) {
+        m_eventPipeNotifier->setEnabled(false);
+        sensordLogW("fd mismatch, event pipe notifier disabled");
+    } else {
+        const size_t maxEvents = 64;
```
Comment (jusa) [inline core/hybrisadaptor.cpp]: These two magic values could be somewhere?

Comment (spiiroin) [reply core/hybrisadaptor.cpp]: Yes, ideally it should be one and same constant.
I started doing that at some stage but gave up as it yielded a lot of diff noise.
Maybe I should do it as a separate commit...

File: core/hybrisadaptor.cpp (hunk 1 above)
Comment (spiiroin) [reply core/hybrisadaptor.cpp]: Fixed

File: core/hybrisadaptor.cpp (hunk 2 above)
Comment (mlehtima) [reply core/hybrisadaptor.cpp]: Just a define at the top of the file should be simple enough.

Comment (spiiroin) [reply core/hybrisadaptor.cpp]: Yeah, but there are other itches to scratch once one starts...

**Review by mlehtima — APPROVED** [review]

**Pull request merged**